lte/gateway/c/core/oai/test/mme_app_task/test_mme_procedures.cpp

```diff
@@ -475,6 +475,119 @@ TEST_F(MmeAppProcedureTest, TestImsiAttachEpsOnlyAirTimeout) {
   EXPECT_EQ(mme_state_p->nb_ue_idle, 0);
 }
 
+TEST_F(MmeAppProcedureTest, TestImsiAttachEpsOnlyAuthMacFailure) {
+  mme_app_desc_t* mme_state_p =
+      magma::lte::MmeNasStateManager::getInstance().get_state(false);
+  std::condition_variable cv;
+  std::mutex mx;
+  std::unique_lock<std::mutex> lock(mx);
+
+  MME_APP_EXPECT_CALLS(3, 0, 2, 1, 0, 0, 0, 0, 0, 0, 1);
+
+  // Construction and sending Initial Attach Request to mme_app mimicing S1AP
+  send_mme_app_initial_ue_msg(
+      nas_msg_imsi_attach_req, sizeof(nas_msg_imsi_attach_req), plmn, guti, 1);
+
+  // Sending AIA to mme_app mimicing successful S6A response for AIR
+  send_authentication_info_resp(imsi, true);
+
+  // Wait for DL NAS Transport for Auth Req
+  cv.wait_for(lock, std::chrono::milliseconds(STATE_MAX_WAIT_MS));
+
+  // Constructing and sending Authentication Response with Mac Failure to
+  // mme_app mimicing S1AP
+  // Message Type 0x5c = Auth Failure
+  // Cause 0x14 = MAC failure
+  const uint8_t nas_msg_auth_resp_mac_fail[19] = {
+      0x07, 0x5c, 0x14, 0x30, 0x0e, 0x00, 0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+
+  send_mme_app_uplink_data_ind(
+      nas_msg_auth_resp_mac_fail, sizeof(nas_msg_auth_resp_mac_fail), plmn);
+
+  // Wait for DL NAS Transport for Auth Reject
+  cv.wait_for(lock, std::chrono::milliseconds(STATE_MAX_WAIT_MS));
+
+  // Wait for context release request
+  cv.wait_for(lock, std::chrono::milliseconds(STATE_MAX_WAIT_MS));
```

Comment:
we are sending context release twice? Can you file an issue to not do that?

Reply:
Agreed, added comment in issue https://github.com/magma/magma/issues/9969 to audit all common procedure failure call flows.